config/methods.config

```diff
@@ -99,7 +138,19 @@ methods {
         if (params.multi_tumor_sample || params.multi_normal_sample) {
             params.sample_id = params.patient_id
         } else {
-            params.sample_id = params.input['tumor'][0]['id']
+            def tumorSample = params.samples_to_process.find { it['sample_type'] == 'tumor' }
+            if (tumorSample) {
+                params.sample_id = tumorSample['id']
+                params.tumor_id = params.sample_id
+            } else {
+                throw new Exception("Warning: No sample with sample_type 'tumor' found in ${params.samples_to_process}")
+            }
+            if (params.tumor_only_mode) {
+                params.normal_id = 'Empty_id'
+            } else {
+                params.normal_id = params.samples_to_process.find { it['sample_type'] == 'normal' }['id']
+                params.single_NT_paired = true
+            }
```

Comment:
This set_sample_params function is starting to get fairly long and complicated; it may be potentially simplified with something along the lines of: get the counts and IDs of normal and tumor samples then handle setting different parameters as needed

Reply:
done